src/overlays/actors/ovl_Bg_Ydan_Sp/z_bg_ydan_sp.c

```diff
@@ -63,29 +73,377 @@ static ColliderTrisInit D_808C09A8 = {
+        this->unk16C = 0.0f;
+    } else {
+        CollisionHeader_GetVirtual(&D_06006050, &colHeader);
+        this->actionFunc = &BgYdanSp_WallWebIdle;
+        Actor_SetFocus(thisx, 30.0f);
+        sinsY = Math_SinS(thisx->shape.rot.y);
+        cossY = Math_CosS(thisx->shape.rot.y);
+        nSinsX = -Math_SinS(thisx->shape.rot.x);
+        cossX = Math_CosS(thisx->shape.rot.x);
+
+        for (i = 0; i < 3; i++) {
+            tri[i].x = thisx->world.pos.x + (cossY * ti1->dim.vtx[i].x) - (sinsY * ti1->dim.vtx[i].y * nSinsX);
+            tri[i].y = thisx->world.pos.y + (ti1->dim.vtx[i].y * cossX);
+            tri[i].z = thisx->world.pos.z - (sinsY * ti1->dim.vtx[i].x) + (ti1->dim.vtx[i].y * cossY * nSinsX);
+        }
+
+        Collider_SetTrisVertices(&this->trisCollider, 0, &tri[0], &tri[1], &tri[2]);
+
+        tri[1].x = thisx->world.pos.x + (cossY * ti1->dim.vtx[0].x) - (ti1->dim.vtx[2].y * sinsY * nSinsX);
+        tri[1].y = thisx->world.pos.y + (ti1->dim.vtx[2].y * cossX);
+        tri[1].z = thisx->world.pos.z - (sinsY * ti1->dim.vtx[0].x) + (ti1->dim.vtx[2].y * cossY * nSinsX);
+        Collider_SetTrisVertices(&this->trisCollider, 1, &tri[0], &tri[2], &tri[1]);
+    }
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    this->timer = 0;
+    if (Flags_GetSwitch(globalCtx, this->isDestroyedSwitchFlag)) {
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void BgYdanSp_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanSp* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(globalCtx, &this->trisCollider);
+}
+
+void* BgYdanSp_UpdateFloorWebCollision(BgYdanSp* this) {
+    s16 newY;
+    CollisionHeader* colHeader;
+
+    colHeader = (CollisionHeader*)SEGMENTED_TO_VIRTUAL(&D_06006460);
+    colHeader->vtxList = (Vec3s*)SEGMENTED_TO_VIRTUAL(colHeader->vtxList);
+    newY = (s16)((this->dyna.actor.home.pos.y - this->dyna.actor.world.pos.y) * 10.0f);
+    colHeader->vtxList[14].y = newY;
+    colHeader->vtxList[12].y = newY;
+    colHeader->vtxList[10].y = newY;
+    colHeader->vtxList[9].y = newY;
+    colHeader->vtxList[6].y = newY;
+    colHeader->vtxList[5].y = newY;
+    colHeader->vtxList[1].y = newY;
+    colHeader->vtxList[0].y = newY;
+}
+
+void BgYdanSp_BurnWeb(BgYdanSp* this, GlobalContext* globalCtx) {
+    this->timer = 30;
+    this = this;
+    func_80078884(NA_SE_SY_CORRECT_CHIME);
+    Flags_SetSwitch(globalCtx, this->isDestroyedSwitchFlag);
+    if (this->dyna.actor.params == WEB_FLOOR) {
+        this->actionFunc = &BgYdanSp_BurnFloorWeb;
+        return;
+    }
+    this->actionFunc = &BgYdanSp_BurnWallWeb;
+}
+
+void BgYdanSp_BurnFloorWeb(BgYdanSp* this, GlobalContext* globalCtx) {
+    static Vec3f accel = { 0 };
+    Vec3f velocity;
+    Vec3f pos2;
+    f32 distXZ;
+    f32 sins;
+    f32 coss;
+    s16 rot;
+    s16 rot2;
+    s32 i;
+
+    DECR(this->timer);
```

Comment:
DECR shouldn't be used on its own ideally. Relevant in other functions in this file
```suggestion
    if (this->timer != 0) {
        this->timer--;
    }
```